vumi/transports/smpp/tests/test_protocol.py

```diff
@@ -1,74 +1,76 @@
+    Unbind, UnbindResp, SubmitSMResp, DeliverSM, EnquireLink)
 from vumi.tests.helpers import VumiTestCase, PersistenceHelper
-from vumi.transports.smpp.smpp_transport import SmppTransceiverTransport
+from vumi.transports.smpp.smpp_transport import (
+    SmppTransceiverTransport, SmppMessageDataStash)
 from vumi.transports.smpp.protocol import (
     EsmeProtocol, EsmeProtocolFactory)
 from vumi.transports.smpp.pdu_utils import (
     seq_no, command_status, command_id, short_message)
 from vumi.transports.smpp.smpp_utils import unpacked_pdu_opts
 from vumi.transports.smpp.sequence import RedisSequence
-from vumi.transports.smpp.tests.helpers import bind_protocol, wait_for_pdus
-from vumi.transports.tests.helpers import TransportHelper
-
-from smpp.pdu import unpack_pdu
-from smpp.pdu_builder import (
-    Unbind, UnbindResp, SubmitSMResp, DeliverSM, EnquireLink, EnquireLinkResp)
-
+from vumi.transports.smpp.tests.fake_smsc import FakeSMSC
 
-def connect_transport(protocol):
-    transport = proto_helpers.StringTransport()
-    protocol.makeConnection(transport)
-    return transport
 
+class DummySmppTransport(object):
+    def __init__(self, clock, redis, config):
+        self.clock = clock
+        self.redis = redis
+        self._config = config
+        self._static_config = SmppTransceiverTransport.CONFIG_CLASS(
+            self._config, static=True)
 
-class ForwardableRedisSequence(RedisSequence):
+        config = self.get_static_config()
+        self.dr_processor = config.delivery_report_processor(
+            self, config.delivery_report_processor_config)
+        self.deliver_sm_processor = config.deliver_short_message_processor(
+            self, config.deliver_short_message_processor_config)
+        self.submit_sm_processor = config.submit_short_message_processor(
+            self, config.submit_short_message_processor_config)
+        self.sequence_generator = RedisSequence(self.redis)
+        self.message_stash = SmppMessageDataStash(self.redis, config)
 
-    def advance(self, seq_nr):
-        d = self.redis.set('smpp_last_sequence_number', seq_nr)
-        d.addCallback(lambda _: seq_nr)
-        return d
+        self.paused = True
 
+    def get_static_config(self):
+        return self._static_config
 
-class DummySmppTransport(SmppTransceiverTransport):
+    def pause_connectors(self):
+        self.paused = True
 
-    sequence_class = ForwardableRedisSequence
+    def unpause_connectors(self):
+        self.paused = False
 
 
 class TestEsmeProtocol(VumiTestCase):
 
     @inlineCallbacks
     def setUp(self):
-        self.tx_helper = self.add_helper(TransportHelper(DummySmppTransport))
+        self.clock = Clock()
         self.persistence_helper = self.add_helper(PersistenceHelper())
         self.redis = yield self.persistence_helper.get_redis_manager()
-        self.clock = Clock()
-        self.patch(EsmeProtocol, 'clock', self.clock)
+        self.fake_smsc = FakeSMSC(auto_accept=False)
```

Comment:
The main reason to do that cleanup is to avoid leaving the reactor dirty at the end of the test. Since we're using a fake clock and a fake endpoint, there's nothing in the reactor that needs cleaning up.